packages/core/src/config-loader/config-loader.ts

```diff
@@ -99,10 +99,18 @@ export class ConfigLoader {
   }
 
   async reload() {
-    // make sure everything is initialized
-    await this.isReady;
+    if (!this.workspaceInfo) await this.finishInit();
 
-    if (!this.viteRunner) throw new Error('vite server not ready yet');
+    // make sure everything is initialized
+    // await this.isReady;
+
+    if (!this.viteNodeRunner) {
```

Comment:
moving the vite server init is definitely going to make things slower, so we should try to move it and make sure it's happening as early as possible.